src/lsp/main.ts

```diff
@@ -358,20 +359,18 @@ const generalCommands = [
     },
 ];
 
-function registerCommands(
-    context: vscode.ExtensionContext,
-    client: LanguageClient
-) {
-    context.subscriptions.push(
-        ...generalCommands.map((command) =>
-            vscode.commands.registerCommand(command.name, command.handler)
-        )
-    );
-    context.subscriptions.push(
-        ...clojureLspCommands.map((command) =>
-            registerLspCommand(client, command)
-        )
-    );
+function registerLspCommands(context: vscode.ExtensionContext) {
+    if (!lspCommandsRegistered) {
+        context.subscriptions.push(
+            ...generalCommands.map((command) =>
+                vscode.commands.registerCommand(command.name, command.handler)
+            )
+        );
+        context.subscriptions.push(
+            ...clojureLspCommands.map((command) => registerLspCommand(command))
+        );
+    }
+    lspCommandsRegistered = true;
```

Comment:
You might also be able to store the disposables when registering and dispose of the commands when the server is stopped, which might be better than leaving them registered, but that might not be super important and can be done later.